packages/dma-contracts/contracts/core/OperationExecutor.sol

```diff
@@ -1,120 +1,131 @@
+    CHAINLOG_VIEWER = _chainLogView;
+    BALANCER_VAULT = _balanacerVault;
+    OPERATION_EXECUTOR = address(this);
   }
 
   /**
-   * @notice Executes an operation
-   * @dev
-   * There are operations stored in the OperationsRegistry which guarantee the order of execution of actions for a given Operation.
-   * There is a possibility to execute an arrays of calls that don't form an official operation.
+   * @notice Executes a list of action calls that form an Operation
+   * @dev Call to this contract MUST be done through a Proxy contract.
+   * Either through a https://github.com/makerdao/ds-proxy
+   * or https://github.com/OasisDEX/defi-position-manager-smartcontracts/blob/master/contracts/AccountImplementation.sol
+   * or something similar.
    *
-   * Operation storage is cleared before and after an operation is executed.
+   * During the transaction execution there is a proxy storage slot reserved that keeps
+   * all the data that's used during the transaction execution.
+   * After transaction completes, the data in that slot is deleted.
    *
-   * To avoid re-entrancy attack, there is a lock implemented on OpStorage.
-   * A standard reentrancy modifier is not sufficient because the second call via the onFlashloan handler
-   * calls aggregateCallback via DSProxy once again but this breaks the special modifier _ behaviour
-   * and the modifier cannot return the execution flow to the original function.
-   * This is why re-entrancy defence is immplemented here using an external storage contract via the lock/unlock functions
-   * @param calls An array of Action calls the operation must execute
-   * @param operationName The name of the Operation being executed
+   * During the execution each action hash is added to a common variable and then
+   * the whole thing is hashed using keccak256.
+   * Then using that hased value there is a lookup in the OperationsRegistry for a specific operation
+   * that corresponds to that value, If there isn't, the transaction reverts.
+   *
+   * @param calls List of action calls to be executed.
    */
-  function executeOp(Call[] memory calls, string calldata operationName) public payable {
-    OperationStorage opStorage = OperationStorage(registry.getRegisteredService(OPERATION_STORAGE));
-    opStorage.lock();
-    OperationsRegistry opRegistry = OperationsRegistry(
-      registry.getRegisteredService(OPERATIONS_REGISTRY)
-    );
+  function executeOp(Call[] memory calls) public payable {
+    StorageSlot.TransactionStorage storage txStorage = StorageSlot.getTransactionStorage();
+    delete txStorage.actions;
+    delete txStorage.returnedValues;
 
-    opStorage.clearStorage();
-    (bytes32[] memory actions, bool[] memory optional) = opRegistry.getOperation(operationName);
-    opStorage.setOperationActions(actions, optional);
     aggregate(calls);
+    bytes32 operationName = this.getOperation(keccak256(abi.encodePacked(txStorage.actions)));
```

Comment:
is `.this` necessary ?

Reply:
that won't work I think, if the `executeOp` is delegatecall'ed - as `this` will be the proxy